nexus/src/app/sagas/instance_start.rs

```diff
@@ -426,7 +431,7 @@ async fn sis_ensure_registered_undo(
     let osagactx = sagactx.user_data();
     let params = sagactx.saga_params::<Params>()?;
     let datastore = osagactx.datastore();
-    let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
```

Comment:
For my own edification: The removal of this lookup (and also the one on line 374) are the "fixes" for this PR. Everything else is clarity + tests.

Reply:
Correct, this part is the fix; everything else is fixing the tests and making the logs a little easier to understand. (Probably shouldn't have squashed the commits.) With only the test fixes in place, the unwind test fails because of a panic during undo that matches what we saw in dogfood.